server/api.go

```diff
@@ -45,6 +46,30 @@ type dexAPI struct {
 	version string
 }
 
+func (d dexAPI) GetClient(ctx context.Context, req *api.GetClientReq) (*api.GetClientResp, error) {
+	sanitizedID := strings.Trim(req.Id, " \t\n")
+	if sanitizedID == "" {
+		return nil, errors.New("get client: no client ID supplied")
+	}
```

Comment:
I wonder why sanitization needs to happen here.

Reply:
in theory, someone could send a string like "  my-id ", this way we keep stuff clean.

Reply:
Ping? What do you want me to do? Can we merge this one? Do you want me to change something?

Reply:
Hello, @marcofranssen. Let's go without sanitizing ids for now. If a user sends an id, it must be the exact id without trailing spaces.

Reply:
ok, that is exactly what the sanitizing does. It removes all trailing and leading whitespacing.